### PR TITLE
Migrate bicep example: 101/sqlmi-new-vnet

### PR DESCRIPTION
**Pull request by StephenWeatherford**
Added bicep support to this sample by integrating bicep.main from the example in https://github.com/Azure/bicep/tree/main/docs/examples/101/sqlmi-new-vnet

The corresponding PR for the modified bicep example is here: https://github.com/Azure/bicep/pull/3244

### CODE REVIEW

**Comment by azure-quickstarts** [comment]
@stevestein - check this PR for updates that may be needed to documentation that references this sample.  [This is an automated message. You are receiving it because you are listed as the docOwner in metadata.json.]

**Comment by StephenWeatherford** [comment]
FYI @srdan-bozovic-msft

**Comment by azure-quickstarts** [comment]
@stevestein - check this PR for updates that may be needed to documentation that references this sample.  [This is an automated message. You are receiving it because you are listed as the docOwner in metadata.json.]

**Comment by azure-quickstarts** [comment]
@stevestein - check this PR for updates that may be needed to documentation that references this sample.  [This is an automated message. You are receiving it because you are listed as the docOwner in metadata.json.]

File: quickstarts/microsoft.sql/sqlmi-new-vnet/main.bicep [hunk 1]
```diff
@@ -0,0 +1,187 @@
...
+  }
+  properties: {
+    administratorLogin: administratorLogin
+    administratorLoginPassword: administratorLoginPassword
+    subnetId: resourceId('Microsoft.Network/virtualNetworks/subnets', virtualNetworkName, subnetName)
+    storageSizeInGB: storageSizeInGB
+    vCores: vCores
+    licenseType: licenseType
+  }
+  dependsOn: [
```
Comment (bmoore-msft) [inline quickstarts/microsoft.sql/sqlmi-new-vnet/main.bicep:184]: move this above properties

**Review by bmoore-msft — CHANGES_REQUESTED** [review]

**Comment by azure-quickstarts** [comment]
@stevestein - check this PR for updates that may be needed to documentation that references this sample.  [This is an automated message. You are receiving it because you are listed as the docOwner in metadata.json.]

**Comment by azure-quickstarts** [comment]
@stevestein - check this PR for updates that may be needed to documentation that references this sample.  [This is an automated message. You are receiving it because you are listed as the docOwner in metadata.json.]

**Pull request merged**